Integrate abstractions projects into client build.

MaxKsyunz opened the pull request:
- Added projects under `abstractions\src` to `OpenSearch.sln`.
- Removed build scripts specific to abstractions -- they are not needed any more.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Integrate abstractions projects into client build.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

MaxKsyunz commented on Directory.Build.targets:
```diff
@@ -1,8 +1,8 @@
+  <PropertyGroup Condition="$(SignAssembly) == True OR $(IsPackable) == True OR $(MSBuildProjectName.StartsWith('Tests'))">
     <SignAssembly>true</SignAssembly>
-    <AssemblyOriginatorKeyFile>$(SolutionDir)\build\keys\keypair.snk</AssemblyOriginatorKeyFile>
+    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)\build\keys\keypair.snk</AssemblyOriginatorKeyFile>
```
`$(SolutionDir)` is only defined inside an IDE, `$(MSBuildThisFileDirectory)` is defined by `msbuild`.

Relative path would have worked as well but may look ambiguous.

Yury-Fridlyand commented:
Can you add misc files to the solution as well:
```
abstractions/*
abstractions/src/Directory.Build.props
```

MitchellGale commented on OpenSearch.sln:
```diff
@@ -12,10 +12,10 @@ EndProject
+	ProjectSection(SolutionItems) = preProject
+		tests\Directory.Build.props = tests\Directory.Build.props
+		tests\.runsettings = tests\.runsettings
+	EndProjectSection
```
are the new intents needed?

MaxKsyunz replied:
These were done by the `dotnet` cli. 

It looks like `ProjectSection` is an element under `Project` and so needs an indent.

forestmvey commented on OpenSearch.sln:
```diff
@@ -12,10 +12,10 @@ EndProject
 Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "scripts", "build\scripts\scripts.fsproj", "{D6997ADC-E933-418E-831C-DE1A78897493}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "tests", "tests", "{6C4A2627-AF22-4388-9DF7-7A9AEACFD635}"
-ProjectSection(SolutionItems) = preProject
```
What's the purpose to these whitespace changes in this file?

MaxKsyunz replied:
These were done by `dotnet` cli. 

It looks like `ProjectSection` is an element under `Project` and so needs an indent.

forestmvey commented on abstractions/README.md:
```diff
@@ -49,29 +44,3 @@ Supports:
 2. Released versions
     * `MAJOR.MINOR.PATH` where `MAJOR` is still supported as defined by the EOL policy of OpenSearch.
     * Note if the version exists but is not yet released it will resolve as a build candidate
-    
```
Curious why this is removed from the readme?

MaxKsyunz replied:
It was asked for in the [upstream PR](https://github.com/opensearch-project/opensearch-net/pull/92#discussion_r961923490).

MaxKsyunz commented:
@Yury-Fridlyand, as per your [comment](https://github.com/Bit-Quill/opensearch-net/pull/41#issuecomment-1235841737),  I added `abstractions\README.md` and removed `Directory.Build.props` because it was functionally empty.
